app/portainer/settings/edge-compute/SettingsEdgeCompute/SettingsEdgeCompute.tsx

```diff
@@ -0,0 +1,128 @@
+import { Formik, Form } from 'formik';
+
+import { Switch } from '@/portainer/components/form-components/SwitchField/Switch';
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { Select } from '@/portainer/components/form-components/Input/Select';
+import { Widget, WidgetBody, WidgetTitle } from '@/portainer/components/widget';
+import { LoadingButton } from '@/portainer/components/Button/LoadingButton';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+
+import styles from './SettingsEdgeCompute.module.css';
+
+export interface FormValues {
+  EdgeAgentCheckinInterval: number;
+  EnableEdgeComputeFeatures: boolean;
+}
+
+interface Props {
+  settings: FormValues;
+  onSubmit(values: FormValues): void;
+}
+
+const checkinIntervalOptions = [
+  {
+    value: 5,
+    label: '5 seconds',
+  },
+  {
+    value: 10,
+    label: '10 seconds',
+  },
+  {
+    value: 30,
+    label: '30 seconds',
+  },
+];
+
+export function SettingsEdgeCompute({ settings, onSubmit }: Props) {
+
+  const initialValues = {
+    EdgeAgentCheckinInterval: settings ? settings.EdgeAgentCheckinInterval : 5,
+    EnableEdgeComputeFeatures: settings ? settings.EnableEdgeComputeFeatures : false,
+  };
+
+  return (
+    <div className="row">
+      <Widget>
+        <WidgetTitle icon="fa-laptop" title="Edge Compute settings" />
+        <WidgetBody>
+          <Formik
+            initialValues={initialValues}
+            enableReinitialize
+            onSubmit={onSubmit}
+            validateOnMount
```

Comment:
add validation schema

Reply:
it will be actually empty, but I'll add it in case this form grows in the future